source/integrations/mongoose-get-started.txt

```diff
@@ -0,0 +1,642 @@
+matches the provided query. The following is an example of matching an article
+based on the ``author`` field:
+
+.. code-block:: javascript
+
+   const blog = await Blog.exists({ author: 'Jess Garcia' });
+   console.log(blog);
+
+where()
+~~~~~~~
+
+Mongoose has a unique style of querying data. The ``where()`` method allows you to
+chain and build queries. The following is an example of performing a find
+operation using ``findOne()`` and the equivalent approach using the ``where()`` method:
+
+.. code-block:: javascript
+
+   const blogFind = await Blog.findOne({ author: "Jess Garcia" });
+   console.log(blogFind);
+
+   const blogWhere = await Blog.where("author").equals("Jess Garcia");
+   console.log(blogWhere);
+
+You can use either method to perform find operations. You can also chain
+multiple ``where()`` methods to build increasingly complex queries.
+
+To include projection when using the ``where()`` method, chain the ``select()`` method
+after your query, as shown in the following example:
+
+.. code-block:: javascript
+
+   const blog = await Blog.where("author").equals("Jess Garcia").select("title author");
+   console.log(blog);
+
+
+Validate Your Data
+------------------
+
+The articles inserted so far have not contained the ``author``, ``dates``, or
+``comments`` fields, even though these fields are included in the schema. This is
+because although you defined the structure of your document, you have not
+defined which fields are required. Without defining the required fields, you can
+omit any fields.
+
+To add data validation and define these requirements, update the schema in
+``Blog.js`` like the following:
+
+.. literalinclude:: /includes/integrations/mongoose-blogSchema-validate.js
+   :language: javascript
+   :start-after: start-blogSchema
+   :end-before: end-blogSchema
+   :dedent:
+
+In Mongoose, when you include validation on a field, you must pass an object as
+its value. 
+
+..  tip::
+
+    When you use schemas with Mongoose, ``value: String`` is the same as
+    ``value: {type: String}``.
+
+You can use several validation methods with Mongoose. For example, you can set
+``required`` to true on any fields that you want to require.
+
+You can also validate the type and the formatting. In the preceding code, the
+``slug`` field is defined as a ``string`` that is always in ``lowercase``. This
+validation takes the slug input and converts it to lowercase before saving the
+document to the database.
+
+For the ``createdAt`` date field, you can set the default by using an arrow function.
+This field is also specified to be impossible to change later by setting
+``immutable`` to ``true``.
+
+.. note::
+
+   Validators only run on the ``create()`` and ``save()`` methods.
+
+Multiple Schemas
+----------------
+
+Now that you have validation on your blog schema, and the author field is
+``required``, you must update ``index.js`` to include the author. To do this,
+you can create a separate schema.
```

Comment:
Similar to my earlier comment - this intro paragraph doesn't make as much sense when not in the context of a stepped tutorial. For this one I think it makes sense not to be in a tutorial though, so maybe just a rewording to make it more of a general intro paragraph would be helpful